internal/querynode/query_collection.go

```diff
@@ -1069,6 +1069,12 @@ func (q *queryCollection) search(msg queryMsg) error {
 }
 
 func (q *queryCollection) fillVectorFieldsData(segment *Segment, result *segcorepb.RetrieveResults) error {
+	if segment.segmentType == segmentTypeGrowing ||
+		(segment.segmentType == segmentTypeSealed && len(segment.indexInfos) == 0) {
```

Comment:
If there are multiple vector columns in the segment, you may need to use fieldID to check for the presence of the corresponding vector field.